libsolidity/codegen/MultiUseYulFunctionCollector.cpp

```diff
@@ -46,7 +46,9 @@ string MultiUseYulFunctionCollector::requestedFunctions()
 
 string MultiUseYulFunctionCollector::createFunction(string const& _name, function<string ()> const& _creator)
 {
-	if (!m_requestedFunctions.count(_name))
+	if (string* existingFunction = valueOrNullptr(m_requestedFunctions, _name))
+		solAssert(*existingFunction == _creator(), "");
```

Comment:
I think this can lead to exponential runtime behaviour.

Reply:
Do we have recursions in these? The ``STUB`` stuff actually looks like it... then it might not even terminate...